CourseStream/CourseStream.UnitTests/CourseStreamServiceTest.cs

```diff
@@ -126,7 +126,8 @@ public async Task ListAsync_ReturnsMappedCourseStreams()
         public void CourseStreamService_Constructor_InitializesRepository()
         {
             // Arrange & Act
-            var service = new CourseStreamService();
+            var mockRepo = new Mock<ICourseStreamRepository>();
```

Comment:
Define mocks at the class level. Not at the methods. Each method is an isolated test and mocks have to be cleaned up by the framework (xUnit). Here confirmation required. I haven't write tests for quiet a long and can miss some important aspects